src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnection.cs

```diff
@@ -972,7 +974,35 @@ private async ValueTask SendRequestContentAsync(HttpRequestMessage request, Http
             }
         }
 
-        private static void ParseStatusLine(ReadOnlySpan<byte> line, HttpResponseMessage response)
+        private bool ParseStatusLine(HttpResponseMessage response)
+        {
+            Span<byte> buffer = _readBuffer.AsSpan(_readOffset, _readLength - _readOffset);
```

Comment:
Were you planning to update this to remember the last place it started a search from in the case of needing to retry?  Or did you decide it's not an issue?

Reply:
I ended up adding such logic in the [`FillForHeadersAsync`](https://github.com/dotnet/runtime/blob/b62e86bcddd798ca226a1118224ad6a5afc208d1/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnection.cs#L1755) helper that we call instead of `FillAsync` when reading headers.
The helper has a slow path that's only used if we just did a `FillAsync` call but haven't consumed a single byte. It guarantees `O(n)` worst-case performance.

This logic is reused by `ChunkedEncodingReadStream`'s trailing header parsing.

I also added some pathological test cases (slowly trickled huge header, millions of line folds in the same header ...)